test/3rdparty/concurrencytest-0.1.2/concurrencytest.py

```diff
@@ -43,6 +45,23 @@
 
 CPU_COUNT = cpu_count()
 
+def wait_for_children(pids):
```

Comment:
It's possible that this would be the first time we'd diverged from the third party code - if that's the case, we should think carefully about it (i.e: ensure we don't loose the change if we ever decided to upgrade to a new version), and probably attempt to send the change upstream as well.

Reply:
Right, I'm not sure what our process for this is. Hoping Don can provide some guidance. I worked at it to limit the changes to this file only in terms of 3rd party changes. We did see this project (concurrencytest) is not active.

Reply:
Yes, I'm pretty sure this is the first time we've diverged from Python third party code.  To make it clear to future maintainers, I'd do three things.  First, I'd rename `test/3rdparty/[concurrencytest-0.1.2/` to be something like `test/3rdparty/[concurrencytest-0.1.2-locally-modified/` .  The `run.py` doesn't care what the directory names are, so I believe this will continue to work everywhere.  Second, I'd throw a 1 line README in this directory, to make sure it catches anybody's attention.  Third, I'd put a comment here (in this file) alerting to the modification (sure it's in git, but....).

Reply:
Updated in latest push.